### ISSUE TITLE
Adding a slider fails to load visualization in version 0.12.14

### ISSUE THREAD

**Issue by seanlaw** [bokeh/bokeh#7591]
I'm running the simple test case below using a bokeh server:

```
from bokeh.plotting import figure, curdoc
from bokeh.layouts import layout
from bokeh.models import Slider

plot = figure()
plot.circle([1,2,3], [1,2,3])

def get_layout():
    l = layout([[plot], [Slider()]], sizing_mode='stretch_both')
    return l

curdoc().add_root(get_layout())

```

```
$ bokeh serve --show test.py
```
It loads fine in the browser (Chrome) with bokeh 0.12.6 and all the way up version 0.12.13 installed via 

```
conda install bokeh=0.12.13
```
(build channel py36_0) but the same example above fails to load with bokeh version 0.12.14

I suspect that there was some breaking changes that happened in this version.

**Comment by bryevdv** [comment]
@seanlaw this example displays if I make a slider with values:
```
Slider(start=0, end=10, value=1)
``` 
Without this, I see a browser error:
```
Error: noUiSlider (10.1.0): 'range' 'min' and 'max' cannot be equal.
```
We did change slider implementations at some point. It's possible the new slider is less lax about some some things. I'm guessing before it made a slider from 0 to 0? In any case I am not sure `Slider()` with no arguments was ever intended usage or worked as more than an accident.

**Comment by seanlaw** [comment]
It looks like the Slider() was a red herring. I'm using Bokeh in a more elaborate dashboard and am trying to debug with the simplest case. Sorry for the confusion. I think I've made progress with this code:

```
from bokeh.plotting import figure, curdoc
from bokeh.layouts import layout
from bokeh.models import Slider

plot = figure()  # This renders the plot but it is squashed vertically and only stretches when you drag the plot
#plot = figure(title='Title')  # This fails to render the plot
#plot = figure(toolbar_location='above')  # This fails to render the plot 
#plot = figure(tools=['box_select', 'reset', 'save', 'tap'])  # This renders the plot but it is squashed vertically and only stretches when you drag the plot

plot.circle([1,2,3], [1,2,3])

s = Slider(start=0.0, end=10.0, value=0)

def get_layout():
    l = layout([[plot], [s]], sizing_mode='stretch_both')
    return l

curdoc().add_root(get_layout())
```

Using ```figure``` without any parameters renders the plot but it is squashed vertically:

<img width="1440" alt="screen shot 2018-03-04 at 8 55 08 pm" src="https://user-images.githubusercontent.com/7473521/36954234-9a7d8bc8-1fee-11e8-96fd-b9c5fdc27253.png">

Adding ```title``` and/or ```toolbar_location``` shows no server-side errors but the client complains with:

```
Error rendering Bokeh items  Error: unsatisfiable constraint: SidePanel(ECE65C2A3B2E47E59B19042FEAE29124, above).height - 30 >= 0
    at t.add_constraint (solver.js:70)
    at Object.i.update_panel_constraints (side_panel.js:156)
    at e.update_constraints (plot_canvas.js:655)
    at e.render (plot_canvas.js:703)
    at e._layout (plot_canvas.js:665)
    at e._layout (layout_dom.js:182)
    at e._layout (layout_dom.js:182)
    at e._layout (layout_dom.js:182)
    at e._do_layout (layout_dom.js:166)
    at e.layout (layout_dom.js:149)
```
Similarly, adding ```tools``` renders the plot but the plot is also squashed. What's weird is that dragging the plot somehow forces the plot to resize automatically to half of the browser height:

<img width="1440" alt="screen shot 2018-03-04 at 8 58 57 pm" src="https://user-images.githubusercontent.com/7473521/36954291-ef30c612-1fee-11e8-8622-45aef1bb02a9.png">

**Comment by seanlaw** [comment]
This issue still exists for 0.12.16 (i.e., the plot is squashed) but there are no client-side errors. Are you able to reproduce this result or is this strictly on my end?

**Comment by bryevdv** [comment]
On master it looks like this:

<img width="1093" alt="screen shot 2018-06-13 at 10 04 51" src="https://user-images.githubusercontent.com/1078448/41366529-4500ba56-6ef1-11e8-9961-2c76f60a1c0a.png">

can you try a `0.13` dev build? Also note that the templating PR was merged so it is now possible to template individual app roots in different divs (e.g. in a booststrap layout instead of  bokeh layout)

**Comment by seanlaw** [comment]
The dev build completely fixed the problem (even for my real use case)!

**Comment by bryevdv** [comment]
Happy Surprise! I think some of the work @mattpap did on some other issues turned out to be related and fixed this coincidentally. Not happy to close without tests but a dedicated effort to get UI/Interaction tests working is on the horizon.

**Comment by mattpap** [comment]
I guess this was fixed in PR #7986.
